### DISCUSSION TITLE
Optimal bot configuration

### DISCUSSION THREAD

**Discussion by adedade** [yuichiroaoki/poly-flashloan-bot#56] [General]
I am trying to figure out what is an optimal configuration, below I want to share some thoughts:

- `loanAmount` what would be a good amount here? Why don't we set it $1M so that the margin are higher, is there a limit in what is the max loan that we can take from DODO?
- What do we think are the best pairs? So far I haven't had a lot of luck with DAI as the base token.
- `routeParts` if I understand the meaning of this properly, should we limit the amount of parts (e.g. max 2-3), since the more parts = more gas we spend = less chances to being able to repay the loan

**Comment by yuichiroaoki** [comment]
For loanAmount
https://github.com/yuichiroaoki/poly-flash/wiki/DODO-Flashloan

For routeParts
It means a ratio of how the bot splits trades when 1inch suggests a specific number of routes.

| number of routes  | part |
| ------------- | ------------- |
| 1  | 100%  |
| 2  | 80%, 20%  |
| 3  | 50%, 40%, 10%  |


![image](https://user-images.githubusercontent.com/45054071/158386762-3d8e1db3-0472-40be-baa9-f98ec0c65be3.png)

**Comment by adedade** [comment]
Hey @yuichiroaoki thanks for following up. What about the value for `diffAmount`, I have updated it to `20` but still getting the `Not enough amount to return loan` error. Should this be a % of the amount vs an arbitrary number?
